### PR TITLE
[tempcontrollers.covesion] Covesion Driver

### PR DESCRIPTION
**Pull request by doddgray**
Driver for Covesion OC crystal oven temperature controller. Still needs docstrings. Can you let me know what else I should change before this can be merged?

### CODE REVIEW

File: instrumental/drivers/tempcontrollers/covesion.py [hunk 1]
```diff
@@ -0,0 +1,230 @@
+# -*- coding: utf-8 -*-
```
Comment (natezb) [inline instrumental/drivers/tempcontrollers/covesion.py]: Add a copyright notice here [e.g. Copyright 2017 Dodd Gray]

File: instrumental/drivers/tempcontrollers/covesion.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+#from .. import _get_visa_instrument
+#from .. import InstrumentTypeError
+#from ... import visa
+
+# def _instrument(params):
+#     inst = _get_visa_instrument(params)
+#     # Should add a check of instrument type here. Not sure how to do this now,
+#     # since querying '*IDN?' doesn't work.
+#     return OC(inst)
+_INST_PARAMS = ['OC_visa_address','version']
```
Comment (natezb) [inline instrumental/drivers/tempcontrollers/covesion.py]: Rename `OC_visa_address` to just `visa_address`.

Also, what is `version`? Does it ever make sense to open a temp controller by version? If not, I'd say remove it here.

File: instrumental/drivers/tempcontrollers/covesion.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        if addr[0:4]=='ASRL':
+            version = _check_OC(rm,addr)
+            if version:
+                params = ParamSet(OC,OC_visa_address=addr,version=version)
+                instruments.append(params)
+    #rm.close()
+    return instruments
+
+
+class OC(TempController):
```
Comment (natezb) [inline instrumental/drivers/tempcontrollers/covesion.py]: It's probably a good idea to rename this something more detailed, like `CovesionOC`

**Comment by doddgray** [comment]
I added docstrings. Still need to replace the print_statusline calls with regular print calls, and ideally fix the bytestrings and communication settings so that the visa comms work reliably. Should I wait until this gets added to the master branch before creating a branch for the hcphotonics driver (since they share the tempcontrollers folder/files included in this pull request)?

**Comment by natezb** [comment]
Looks ready for merging into master! And yeah, it's probably easiest if you rebase that hcphotonics driver on master after this PR gets merged.

I'm going to try out creating an issue (or a few) for discussing and tracking improvements to this driver. I think that could be a useful pattern for all new drivers that get incorporated.

**Comment by doddgray** [comment]
Oh right, i meant to ask- I changed the class parameter name to the standard 'visa_address' as you suggested. I thought I had originally changed this to something non-standard ('OC_visa_address') because the 'visa_address' parameter name is used by some visa instrument handling functions in a way that I thought wasn't compatible with this Covesion OC. Hopefully not a problem, we'll see.

**Pull request merged**